torch/csrc/cuda/Module.cpp

```diff
@@ -1045,9 +1048,15 @@ static void registerCudaPluggableAllocator(PyObject* module) {
   m.def(
       "_cuda_setCheckpointPoolState",
       [](int device,
-         std::shared_ptr<c10::cuda::CUDACachingAllocator::AllocatorState> pps) {
+         std::shared_ptr<c10::cuda::CUDACachingAllocator::AllocatorState> pps,
```

Comment:
Perhaps it would make more sense for setCheckpointPoolState to return `allocated_stale_pointers` and then have this code here do the loop that disables their deleters. This way the allocator doesn't end up StorageImpl objects which are not part of its typical abstraction. It would also help simplify what future pluggable allocators would have to do to checkpoint and restore state. Basically the contract for setCheckpointPoolState will be that it obliterates the state of the allocator and returns a list of live pointers whose deleters need to be handled.

Reply:
Sounds good

Reply:
I made these changes.. 

The one issue here is that now since the storage swapping happens outside of the allocator, there is no way for the allocator to check that the "definitely freed" tensors have storages to swap onto. It will still throw an error in Module.cpp, but then there will be a second error when that tensor dies. Obviously we don't expect to hit this error case, but not ideal.

Reply:
I guess I would probably deliberating on fix in follow up but up to you. For now, I am the only user, and I am checking to make sure that I pass in the right tensors to deallocate.

Reply:
I think for now this is ok, and when the APIs are all in place we can ask what series of actions can lead to this state and figure out the right guards then.